src/js/core/widget/core/DatePicker.js

```diff
@@ -0,0 +1,1844 @@
+/* eslint-disable no-undef */
+/* eslint-disable no-script-url */
+/* eslint-disable camelcase */
+/* eslint-disable guard-for-in */
+/* eslint-disable no-redeclare */
+/* eslint-disable block-scoped-var */
+/* eslint-disable one-var */
+/* eslint-disable vars-on-top */
+/* eslint-disable indent */
+/* eslint-disable no-eq-null */
+/*global window, define, ns */
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Flora License, Version 1.1 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://floralicense.org/license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * @example
+ * <div class="ui-datepicker" id="pickers"></div>
+ *
+ * id="pickers" should be fixed.
+ *
+ * @since 6.0
+ * @class ns.widget.core.DatePicker
+ * @extends ns.widget.BaseWidget
+ * @author Dohyung Lim <delight.lim@samsung.com>
+ */
+(function (window, document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../engine",
+			"../BaseWidget"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var utilsObject = ns.util.object,
+
+				DatePicker = function () {
+					this.options = utilsObject.merge({}, DatePicker.defaults);
+					this.observer = null;
+				},
+
+				defaults = {
+					value: false, // false is past selectable, true is past selection not possible.
+					type: "calendar" // calendar or wheel
+				},
+
+				BaseWidget = ns.widget.BaseWidget,
+				prototype = new BaseWidget();
+
+			DatePicker.prototype = prototype;
+			DatePicker.defaults = defaults;
+
+			prototype._init = function (element) {
+                var TDP;
+
+				if (typeof (TDP) == "undefined") {
+					TDP = {
```

Comment:
What is the "TDP"?
Why are you using the "TDP"?
Can you use the DatePicker Object instead of the TDP?

Reply:
ok. I replace TDP to DatePickerObj.